activerecord/test/cases/enum_test.rb

```diff
@@ -592,6 +592,16 @@ def self.name; "Book"; end
     assert_raises(NoMethodError) { klass.proposed }
   end
 
+  test "scopes are named like methods" do
+    klass = Class.new(ActiveRecord::Base) do
+      self.table_name = "cats"
+      enum breed: { "American Bobtail" => 0, "Balinese-Javanese" => 1 }
```

Comment:
Just curious if user expects generated scopes as `American_Bobtail` and `Balinese_Javanese`, why do not define the enum as `enum breed: { American_Bobtail: 0, Balinese_Javanese: 1 }` from the beginning?

This makes `klass.American_Bobtail` works, but `klass.where(breed: :American_Bobtail)` doesn't work (still be `klass.where(breed: "American Bobtail")`), Isn't it confusing?

As similar thing, we expects that named scope containing space generates as it is:

https://github.com/rails/rails/blob/b91daea9ff73b42a7f6e6479adddf299e60a59c9/activerecord/test/cases/scoping/named_scoping_test.rb#L407-L415

Should we change the expectation too for consistency?

Reply:
> Just curious if user expects generated scopes as American_Bobtail and Balinese_Javanese, why do not define the enum as enum breed: { American_Bobtail: 0, Balinese_Javanese: 1 } from the beginning?

The real-world example where this came up is I have an icon name `dot-fill` and I want to represent that exact value in an `enum` field, so that I can easily get the exact value that my app's icon-renderer expects from the database field. However, I also like having method names that are callable without using metaprogramming like `public_send`, so having a `dot_fill` scope from Rails automatically would be helpful.

> Should we change the expectation too for consistency?

If someone explicitly names their scope with a space in it, I see no problem allowing that. That feels like a separate question than Rails generating a scope on the user's behalf with a space in it, since having a space in a method name doesn't feel common/standard.